api/v1alpha4/openstackcluster_webhook_test.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestOpenStackCluster_ValidateUpdate(t *testing.T) {
+	g := NewWithT(t)
+
+	tests := []struct {
+		name        string
+		oldTemplate *OpenStackCluster
+		newTemplate *OpenStackCluster
+		wantErr     bool
+	}{
+		{
+			name: "OpenStackCluster.Spec.IdentityRef with immutable spec",
```

Comment:
nit: The tests names confused me, and perhaps are still confusing me a little. To me, mutability/immutability is a fundamental property of a thing. 'Immutable' to me means: "no changes at all, ever", whereas this is more like: "this specific change is not allowed".

For this specific test I might have written "Changing OpenStackCluster.Spec.IdentityRef.Kind is not allowed".

This is a nit! Please don't respin for this.

Reply:
Updated the naming. WDYT?